src/tokenserver/extractors.rs

```diff
@@ -0,0 +1,205 @@
+//! Request header/body/query extractors
+//!
+//! Handles ensuring the header's, body, and query parameters are correct, extraction to
+//! relevant types, and failing correctly with the appropriate errors if issues arise.
+
+use actix_web::{dev::Payload, web::Data, Error, FromRequest, HttpRequest};
+use actix_web_httpauth::extractors::bearer::BearerAuth;
+
+use futures::future::LocalBoxFuture;
+use jsonwebtoken::{decode, Algorithm, DecodingKey, Validation};
+
+use crate::server::ServerState;
+use crate::web::error::ValidationErrorKind;
+use crate::web::extractors::RequestErrorLocation;
+
+#[derive(Debug, serde::Serialize, serde::Deserialize, PartialEq)]
+pub struct Claims {
+    pub sub: String,
+    pub iat: i64,
+    pub exp: i64,
+}
+pub struct TokenserverRequest {
+    pub fxa_uid: String,
+}
+
+impl TokenserverRequest {
+    fn get_fxa_uid(jwt: &str, rsa_modulus: String, rsa_exponent: String) -> Result<String, Error> {
+        decode::<Claims>(
```

Comment:
Stupid question, do we know what happens if we send in a JWT with `alg="None"`? That's a fairly common attack vector.

Reply:
Not a stupid question! It looks like the crate we use doesn't support the `None` algorithm, so validation will fail: https://github.com/Keats/jsonwebtoken/blob/2f25cbed0a906e091a278c10eeb6cc1cf30dc24a/src/algorithms.rs#L48-L66

I'll verify this manually to be sure

Reply:
Confirmed that `"alg": "none"` in the JWT headers results in a 400 error